### PR TITLE
[wip] 148034075 cw post foods v2

### PR DESCRIPTION
**Pull request by josh-works**
we're struggling hard with a POST request. Notes in the files.

### CODE REVIEW

File: test/server-test.js [hunk 1]
```diff
@@ -87,6 +87,31 @@ describe('Server', function(){
             })
       })
     })
+
+    describe("POST /foods", function(){
```
Comment (josh-works) [inline test/server-test.js:91]: here's our POST test. It's technically passing, but we're asserting the response from the server is hardcoded to the happy path.

We're working on adding the sad path, but this is a struggle.

File: test/server-test.js [hunk 2]
```diff
@@ -87,6 +87,31 @@ describe('Server', function(){
...
+      it("can create a new food object with properties", function(done){
+        var food = {name: "banana", calories: 50}
+        this.request.post('/api/v1/foods', {form: food}, function(error, response){
+          // console.log(response.body)
+          if(error) {done(error)}
+          var parsedFood = JSON.parse(response.body)
+          assert.equal(response.statusCode, 201)
+          // assert.equal(Foods.countFoods(), 2)
+          assert.equal(parsedFood.status, "success")
+          assert.equal(parsedFood.message, 'Inserted one food')
```
Comment (josh-works) [inline test/server-test.js:110]: two wimpy assertions, pointing over to... one moment.

File: server.js [hunk 3]
```diff
@@ -1,17 +1,36 @@
...
+app.use(bodyParser.json())
+app.use(bodyParser.urlencoded({ extended: true }))
 
+//GET foods
 app.get('/api/v1/foods', function(request, response) {
   Foods.findAll().then(function (data) {
     response.json(data.rows)
   })
 })
+// POST foods
```
Comment (josh-works) [inline server.js:19]: post request path, I believe.

Comment (DBULL7) [reply server.js:19]: It is. is Foods an object you're importing?

File: server.js [hunk 4]
```diff
@@ -39,13 +39,23 @@ app.post('/api/v1/foods', function(request, response) {
...
+
   Foods.find(id).then(function (data) {
       if (data.rowCount == 0) { return response.sendStatus(404) }
       response.json(data.rows[0])
     })
 
 })
 
+// PUT foods/:id
+app.put('/api/vi/foods/:id', function (request, response){
```
Comment (josh-works) [inline server.js]: we're not able to get this function to run (it seems) from the test file.

The `console.log()` on line 52 isn't being hit.

File: test/server-test.js [hunk 5]
```diff
@@ -123,6 +123,30 @@ describe('Server', function(){
...
+      })
+
+      afterEach(function(done){
+        Foods.resetFoods().then(function () { done() })
+      })
+
+      it("can update existing food", function(done){
+        var food = {name: "cheese pizza"}
+
+        this.request.put("api/v1/foods/1", {form: food}, function(error, response){
```
Comment (josh-works) [inline test/server-test.js]: this `put` request should trigger the console.log in the above file. It's not, for some reason.

Any ideas on how to get that console.log (in server.js:52) to run?

@DBULL7 ^^

**Pull request merged**